econml/tests/test_inference.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import numpy as np
+import unittest
+from sklearn.base import clone
+from sklearn.preprocessing import PolynomialFeatures
+from econml.dml import LinearDMLCateEstimator
+
+
+class TestInference(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        np.random.seed(123)
+        # DGP constants
+        cls.n = 1000
+        cls.d_w = 30
+        cls.d_x = 5
+        # Generate data
+        cls.X = np.random.uniform(0, 1, size=(cls.n, cls.d_x))
+        cls.W = np.random.normal(0, 1, size=(cls.n, cls.d_w))
+        cls.T = np.random.normal(0, 1, size=(cls.n, ))
+        cls.Y = np.random.normal(0, 1, size=(cls.n, ))
+
+    def test_inference_results(self):
```

Comment:
I'd rename to something like `test_inference_no_feature_names` or something.

Reply:
I named it this way so we can add more tests under the same method in the future. And this is just one subtest.

Reply:
Okay; I generally prefer to have a larger number of fine grained tests than a smaller number of broader tests where possible because it makes it quicker to track down the nature of the error when there's a failure, but I don't feel super strongly about it.